### DISCUSSION TITLE
How to get the generate command to work with directive parameters?

### DISCUSSION THREAD

**Discussion by stuartcusackie** [spatie/statamic-responsive-images#130] [Q&A] [answered]
I am not using the control panel field. I am just using the blade directive like so:

`@responsive($image, ['ratio' => '2/1', 'lg:ratio' => '3/1'])`

I'm pretty sure this means that the generate command will not create the images needed for this particular view. I ran some tests which confirmed this. It seems that GenerateGlideImageJobs are only created based on the original image dimensions, which in my case I don't need at all. I am always using custom ratios using the method above. The default behaviour is queuing up thousands of jobs for dimensions that my project will never use.

So how would I go about including these ratios in the generation jobs? It would be awesome if the package could scan php files for usages of the tag or directive and add these parameters to the jobs, sort of like how the Tailwind purge works, but that's probably a big ask. 

Config file parameters could work too. e.g.: `'ratios' => ['2/1', '3/1']` but this approach is problematic when we introduce additional parameters into the directive.

Thanks for reading.

**Comment by riasvdv** [comment, marked as answer]
> I'm pretty sure this means that the generate command will not create the images needed for this particular view.

Correct.

> So how would I go about including these ratios in the generation jobs?

There is no way to do this.

As explained in this section: https://github.com/spatie/statamic-responsive-images#using-responsive-images

> We generally recommend setting statamic.assets.image_manipulation.cache to false so only images actually requested by a browser are generated. The first time the conversion is loaded will be slow, but Glide still has an internal cache that it will serve from the next time. This saves a lot on server resources and storage requirements.

There is rarely a good reason to use the manipulation cache with responsive images, it will only lead to unnecessary load & storage.

**Reply by stuartcusackie** [reply]
I had a chat with the Statamic guys yesterday and they explained that some complex logic is ran for each image request when glide caching is turned off, i.e. the server has to determine whether the file exists in cache for a particular manipulation. When caching is turned on the static url of the pre-generated image is simply served and no processing is required. This is why I was curious to experiment with glide caching and the generate glide jobs command, with the hope of some performance boost.

But if this can't be done then no problem. The performance is great either way.

Thanks for clearing this up and big thanks for the package!

**Reply by riasvdv** [reply]
> they explained that some complex logic is ran for each image request when glide caching is turned off

There's been some discussion about solving some of that for Glide 2 (https://github.com/thephpleague/glide/issues/170) but that's probably some way off.
